plan_test.go

```diff
@@ -44,22 +44,34 @@ func TestPlanParams_AppendTo(t *testing.T) {
 		Name:                "Sapphire Elite",
 		StatementDescriptor: "SAPPHIRE",
 	}
-	productId := "prod_123abc"
+	productID := "prod_123abc"
+	tiers := []*PlanTierParams{{Amount: 123, UpTo: 321}, {Amount: 123, UpToInf: true}}
 	testCases := []struct {
 		field  string
 		params *PlanParams
 		want   interface{}
 	}{
 		{"amount", &PlanParams{Amount: 123}, strconv.FormatUint(123, 10)},
+		{"amount", &PlanParams{Amount: 0, AmountZero: true}, strconv.FormatUint(0, 10)},
+		{"amount", &PlanParams{Amount: 0, AmountZero: false}, ""},
+		{"amount", &PlanParams{}, ""},
 		{"currency", &PlanParams{Currency: "USD"}, "USD"},
 		{"id", &PlanParams{ID: "sapphire-elite"}, "sapphire-elite"},
 		{"interval", &PlanParams{Interval: "month"}, "month"},
 		{"interval_count", &PlanParams{IntervalCount: 3}, strconv.FormatUint(3, 10)},
 		{"product[id]", &PlanParams{Product: &productParams}, "ID"},
 		{"product[name]", &PlanParams{Product: &productParams}, "Sapphire Elite"},
 		{"product[statement_descriptor]", &PlanParams{Product: &productParams}, "SAPPHIRE"},
-		{"product", &PlanParams{ProductID: &productId}, "prod_123abc"},
+		{"product", &PlanParams{ProductID: &productID}, "prod_123abc"},
 		{"trial_period_days", &PlanParams{TrialPeriod: 123}, strconv.FormatUint(123, 10)},
+		{"tiers_mode", &PlanParams{TiersMode: "volume"}, "volume"},
+		{"usage_type", &PlanParams{UsageType: "metered"}, "metered"},
+		{"transform_usage[bucket_size]", &PlanParams{TransformUsage: &PlanTransformUsageParams{DivideBy: 123, Round: "round_up"}}, strconv.FormatUint(123, 10)},
+		{"transform_usage[round]", &PlanParams{TransformUsage: &PlanTransformUsageParams{DivideBy: 123, Round: "round_up"}}, "round_up"},
+		{"tiers[0][amount]", &PlanParams{Tiers: tiers}, strconv.FormatUint(123, 10)},
+		{"tiers[0][up_to]", &PlanParams{Tiers: tiers}, strconv.FormatUint(321, 10)},
+		{"tiers[1][amount]", &PlanParams{Tiers: tiers}, strconv.FormatUint(123, 10)},
+		{"tiers[1][up_to]", &PlanParams{Tiers: tiers}, "inf"},
```

Comment:
Sorry for the nit, but would you mind just alphabetizing these test cases by their leading string?